lucene/core/src/java/org/apache/lucene/codecs/blocktree/FieldReader.java

```diff
@@ -84,21 +85,18 @@
     // if (DEBUG) {
     //   System.out.println("BTTR: seg=" + segment + " field=" + fieldInfo.name + " rootBlockCode=" + rootCode + " divisor=" + indexDivisor);
     // }
-
     rootBlockFP = (new ByteArrayDataInput(rootCode.bytes, rootCode.offset, rootCode.length)).readVLong() >>> BlockTreeTermsReader.OUTPUT_FLAGS_NUM_BITS;
-
+    // Initialize FST offheap if index is MMapDirectory and
+    // docCount != sumDocFreq implying field is not primary key
+    isFSTOffHeap = ((this.docCount != this.sumDocFreq) || openedFromWriter == false) && loadFSTOffHeap;
```

Comment:
Should we instead pass in the `FSTLoadMode`  enum down here?  And if it's AUTO then apply these defaulting heuristics?

This way the codec/caller could decide to leave FST off heap even if its a primary key field or its opened from writer?

Reply:
sure we can do that